csharp/ql/src/experimental/ir/implementation/raw/internal/desugar/Foreach.qll

```diff
@@ -359,9 +359,16 @@ private class TranslatedMoveNextEnumAcc extends TTranslatedCompilerGeneratedElem
 
   override Type getResultType() { result instanceof BoolType }
 
+  override Type getVariableType() {
+    exists(TranslatedForeachGetEnumerator ge |
+      ge.getAST() = generatedBy and
+      result = ge.getCallResultType()
+    )
+  }
+
   override predicate hasTempVariable(TempVariableTag tag, CSharpType type) {
     tag = ForeachEnumTempVar() and
-    type = getTypeForPRValue(getResultType())
+    type = getTypeForPRValue(getVariableType())
   }
```

Comment:
I'm not sure this is the best fix. The comment on `hasTempVariable` says that it `Holds if this element generates a temporary variable with type`, but `TranslatedMoveNextEnumAcc` doesn't generate a temp variable, instead it only references one.